test/plugin/test_out_detect_exceptions.rb

```diff
@@ -166,6 +166,11 @@ def test_flush_after_max_lines
     d.run do
       feed_lines(d, t, PYTHON_EXC, JAVA_EXC)
     end
+    # Expected: the firsti two lines of the exception are buffered and combined.
+    # Then the max_lines setting kicks in and the rest Python exception is
+    # logged line-by-line (since it's not an exception stack in itself).
+    # Finally, the Java exception is logged in a single log entry, since it
```

Comment:
Ohh, that explains it.
[Optional] I would propose changing "in a single log entry" to "in its entirety" to make this even clearer.